core/integration/module_test.go

```diff
@@ -6155,6 +6156,716 @@ func (m *Test) PrintDefault(ctx context.Context) (string, error) {
+	return foo.Foo(ctx, value)
+}`
+	const tsSrc = `import { field, func, object } from "@dagger.io/dagger"
+
+  /** @deprecated This module is deprecated and will be removed in future versions. */
+  @object()
+  export class Test {
+    /** @deprecated This field is deprecated and will be removed in future versions. */
+    @field()
+    legacyField = "legacy"
+
+    @func()
+    async echoString(
+	  other: string,
+      /** @deprecated Use 'other' instead of 'input'. */
+      input?: string,
+    ): Promise<string> {
+      return input ?? other
+    }
+
+    /** @deprecated Prefer EchoString instead. */
+    @func()
+    async legacySummarize(note: string): Promise<LegacyRecord> {
+      return { note }
+    }
+
+    @func()
+    useMode(mode: Mode): Mode {
+      return mode
+    }
+
+	@func()
+	async callFoo(foo: Fooer, value: number): Promise<string> {
+		return foo.foo(value)
+	}
+  }
+
+  /** @deprecated This type is deprecated and kept only for retro-compatibility. */
+  export type LegacyRecord = {
+    /** @deprecated This field is deprecated and will be removed in future versions. */
+    note: string
+  }
+
+  export enum Mode {
+    /** @deprecated alpha is deprecated; use zeta instead */
+    Alpha = "alpha",
+    /** @deprecated beta is deprecated; use zeta instead */
+    Beta = "beta",
+    Zeta = "zeta",
+  }
+
+  export interface Fooer {
+    /** @deprecated Use Bar instead */
+    foo(value: number): Promise<string>
+    
+    bar(value: number): Promise<string>
+  }`
+
+	const pySrc = `import enum
+import typing
+from typing import Annotated, Optional
+
+import dagger
+
+@dagger.object_type(
+    deprecated="This module is deprecated and will be removed in future versions."
+)
+class Test:
+    legacy_field: str = dagger.field(
+        name="legacyField",
+        deprecated="This field is deprecated and will be removed in future versions.",
+    )
+
+    @dagger.function(name="echoString")
+    def echo_string(
+        self,
+        input: Annotated[
+            Optional[str], dagger.Deprecated("Use 'other' instead of 'input'.")
+        ],
+        other: str,
+    ) -> str:
+        return input if input is not None else other
+
+    @dagger.function(name="legacySummarize", deprecated="Prefer EchoString instead.")
+    def legacy_summarize(self, note: str) -> "LegacyRecord":
+        return LegacyRecord(note=note)
+
+    @dagger.function(name="useMode")
+    def use_mode(self, mode: "Mode") -> "Mode":
+        return mode
+
+    @dagger.function(name="callFoo")
+    async def call_foo(self, foo: "Fooer", value: int) -> str:
+        return await foo.foo(value)
+
+
+
+@dagger.object_type(
+    deprecated="This type is deprecated and kept only for retro-compatibility."
+)
+class LegacyRecord:
+    note: str = dagger.field(
+        deprecated="This field is deprecated and will be removed in future versions."
+    )
+
+
+@dagger.enum_type
+class Mode(enum.Enum):
+    """Mode is deprecated; use zeta instead."""
```

Comment:
Am I right thinking this line has no impact? We can only deprecate enum values, not enum, right?